src/Campaigns/resources/admin/components/MergeCampaign/Modal/index.tsx

```diff
@@ -1,89 +1,25 @@
-import {useEffect, useState} from 'react';
 import {__} from '@wordpress/i18n';
 import MergeCampaignsForm from './../Form';
-
-/**
- * Remove the "action" query parameter from the current URL
- *
- * @unreleased
- */
-const removeActionParam = () => {
-    const queryParams = new URLSearchParams(window.location.search);
-    const actionParam = queryParams.get('action');
-
-    if (actionParam) {
-        queryParams.delete('action');
-        window.history.replaceState(null, '', `${window.location.pathname}?${queryParams.toString()}`);
-    }
-};
-
-/**
- * Auto open modal if the URL has the query parameter action as "merge"
- *
- * @unreleased
- */
-const autoOpenModal = () => {
-    const queryParams = new URLSearchParams(window.location.search);
-    const actionParam = queryParams.get('action');
-
-    if (actionParam && !window.history.state) {
-        removeActionParam();
-        return false;
-    }
-
-    return actionParam === 'merge';
-};
+import {MergeCampaignModalProps} from '@givewp/campaigns/admin/components/MergeCampaign/Form/types';
 
 /**
  * Create Campaign Modal component
  *
  * @unreleased
  */
-export default function MergeCampaignModal() {
-    const [isOpen, setOpen] = useState<boolean>(autoOpenModal());
+export default function MergeCampaignModal({isOpen, setOpen, campaigns}: MergeCampaignModalProps) {
     const closeModal = () => {
-        removeActionParam();
+        window.history.replaceState('merge-campaigns-modal-closed', '');
         setOpen(false);
     };
```

Comment:
There will be no need for any of this.  You may however need to add additional internal state here for opening & closing the modal that it uses itself.

```js
const [modelIsOpen, setModelIsOpen] = useState(isOpen);

return (
        <>
            <MergeCampaignsForm
                isOpen={modelIsOpen}
                handleClose={()=> setModelIsOpen(false)}
                title={__('Merge campaigns', 'give')}
                campaigns={campaigns}
            />
        </>
    );
```